lib/Zonemaster/CLI.pm

```diff
@@ -407,6 +407,39 @@ sub run {
         Zonemaster::Engine::Profile->effective->merge( $profile );
     }
 
+    my @testing_suite;
+    if ( $self->test and @{ $self->test } > 0 ) {
+        foreach my $t ( @{ $self->test } ) {
+            # There should be at most one slash character
+            if ( $t =~ tr/\/// > 1 ) {
+                die __( "Error: --test must have at most one slash ('/') character: $t");
+            }
+
+            # The case does not matter
+            $t = lc( $t );
+
+            my ( $module, $method );
+            # Fully qualified module and test case (e.g. Example/example12), or just a test case (e.g. example12). Note the different capturing order.
+            if ( ( ($module, $method) = $t =~ m#^ ( [a-z]+ ) / ( [a-z]+[0-9]{2} ) $#ix )
+                or
+                 ( ($method, $module) = $t =~ m#^ ( ( [a-z]+ ) [0-9]{2} ) $#ix ) )
+            {
+                push @testing_suite, "$module/$method";
```

Comment:
Maybe validate the existence of $module/$method in Zonemaster::Engine->all_methods before adding it to @testing_suite? Otherwise if you run `zonemaster-cli --test dnssec05 --test foobar03 example.com`, won't the testing be aborted half way through?